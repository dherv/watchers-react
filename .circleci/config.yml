version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
