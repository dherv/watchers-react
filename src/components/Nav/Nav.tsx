import React, { SFC, MouseEvent, PureComponent } from "react";
import NavItem from "./components/NavItem";
import styles from "./Nav.module.css";

const Nav: SFC<{
  items: string[];
  onClick(item: string): void;
}> = ({ items, onClick }) => (
  <nav>
    <ul className={styles.list}>
      {items.map((item: string) => (
        <NavItem key={item} onClick={onClick} item={item} />
      ))}
    </ul>
  </nav>
);

export default Nav;
