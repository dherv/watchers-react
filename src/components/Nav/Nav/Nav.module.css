.list {
  padding: var(--padding);
  margin: var(--margin);
  box-shadow: var(--shadow);
  border-bottom: 1px solid #4db6ac;
}
