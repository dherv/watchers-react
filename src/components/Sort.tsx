import React, { PureComponent, MouseEvent } from "react";
import styled from "styled-components";

interface IProps {
  sort: (type: string) => void;
}

function Item(props: { children: string; onClick: (type: string) => void }) {
  function handleClick() {
    props.onClick(props.children);
  }
  return <li onClick={handleClick}>{props.children}</li>;
}

export default class Sort extends PureComponent<IProps> {
  render() {
    return (
      <nav>
        <ul>
          <Item onClick={this.props.sort}>date</Item>
          <Item onClick={this.props.sort}>rating</Item>
          <Item onClick={this.props.sort}>name</Item>
        </ul>
      </nav>
    );
  }
}
