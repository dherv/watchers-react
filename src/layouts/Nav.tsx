import React, { PureComponent, MouseEvent } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

interface IProps {}

export default class MainNav extends PureComponent<IProps> {
  render() {
    return (
      <nav>
        <ul>
          <Link to="/movies">movies</Link>
          <Link to="/series">series</Link>
          <Link to="/likes">likes</Link>
          <Link to="/watchlist">watchlist</Link>
          <Link to="/account">account</Link>
        </ul>
      </nav>
    );
  }
}
