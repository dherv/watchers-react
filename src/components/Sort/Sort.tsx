import React, { PureComponent, MouseEvent } from "react";
import styled from "styled-components";
import Nav from "../Nav/Nav";

interface IProps {
  sort: (type: string) => void;
}

export default class Sort extends PureComponent<IProps> {
  constructor(props: IProps) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick(item: string) {
    this.props.sort(item);
  }
  render() {
    return (
      <Nav items={["date", "rating", "name"]} onClick={this.handleClick} />
    );
  }
}
