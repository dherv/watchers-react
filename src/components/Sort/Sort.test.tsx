import { renderer, shallow, React } from "../../test.config";
import Sort from "./Sort";
import Nav from "../Nav/Nav";
describe("Sort", () => {
  const props = {
    sort: jest.fn()
  };
  const wrapper = shallow(<Sort {...props} />);
  describe("component", () => {
    // one Nav
    it("should display one Nav component", () => {
      expect.assertions(2);
      expect(wrapper.find(Nav).length).toBe(1);
      expect(wrapper.find(Nav).prop("items")).toEqual([
        "date",
        "rating",
        "name"
      ]);
    });
    // Nav on click call handleClick
    it("should call handleClick when Nav is clicked", () => {
      expect.assertions(4);
      const handleClick = jest.spyOn(Sort.prototype, "handleClick");
      const wrapper = shallow(<Sort {...props} />);
      const Child = wrapper.find(Nav).first();
      Child.prop("onClick")("item1");
      expect(handleClick).toHaveBeenCalledTimes(1);
      expect(handleClick).toHaveBeenCalledWith("item1");
      expect(props.sort).toHaveBeenCalledTimes(1);
      expect(props.sort).toHaveBeenCalledWith("item1");
    });
    // handleClcik call sort
  });
  describe("snapshot", () => {
    test("should match", () => {
      const tree = renderer.create(<Sort {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
