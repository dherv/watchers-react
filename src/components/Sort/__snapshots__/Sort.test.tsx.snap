// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort snapshot should match 1`] = `
<nav>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      <a
        className="link"
        onClick={[Function]}
      >
        date
      </a>
    </li>
    <li
      className="item"
    >
      <a
        className="link"
        onClick={[Function]}
      >
        rating
      </a>
    </li>
    <li
      className="item"
    >
      <a
        className="link"
        onClick={[Function]}
      >
        name
      </a>
    </li>
  </ul>
</nav>
`;
